src/main/java/org/ice4j/stack/NetAccessManager.java

```diff
@@ -47,6 +51,30 @@
     private static final Logger logger
         = Logger.getLogger(NetAccessManager.class.getName());
 
+    /**
+     * Thread pool to execute {@link MessageProcessingTask}s across all
+     * {@link NetAccessManager}s.
+     */
+    private static ExecutorService messageProcessingExecutor
+        = ExecutorFactory.createFixedThreadPool(
+            Runtime.getRuntime().availableProcessors(),
+            "ice4j.NetAccessManager-");
+
+    /**
+     * Pool of <tt>MessageProcessingTask</tt> objects to avoid extra-allocations
+     * of processor object per <tt>RawMessage</tt> needed to process.
+     */
+    private final ArrayBlockingQueue<MessageProcessingTask> tasksPool
+        = new ArrayBlockingQueue<>(8);
```

Comment:
where does the 8 come from?

Reply:
Initially I've just got it from nowhere, because each  `NetAccessManager` is created per peer and number of `STUN` messages coming from peer not very big. 
Since you've asked I've explicitly verified this by adding `QueueStatistics` to keep track execution of `MessageProcessingTask` tasks.
What I see that even when machine is under load the maximum number of active tasks for single `NetAccessManager` is `2`, so basically pool will have 2 pooled tasks. So 8 seem to be enough pooled tasks with extra safety margin for single `NetAccessManager` instance.